src/backend/parser/parse_coerce.c

```diff
@@ -1380,6 +1381,7 @@ select_common_type(ParseState *pstate, List *exprs, const char *context,
 	TYPCATEGORY pcategory;
 	bool		pispreferred;
 	ListCell   *lc;
+	const char *dump_restore = GetConfigOption("babelfishpg_tsql.dump_restore", true, false);
```

Comment:
I think we are using it at lot of places, may be better to have a backing global variable and use that for check. Also, it should be boolean GUC so values like ON/OFF TRUE/FALSE can all work

Reply:
Actually the guc is defined in tsql extension so we always have to fetch it's value using `GetConfigOption` function because if tsql extension is not installed then the guc will not even be defined. `extern` global variable will create engine's dependency on extension.

Reply:
not for this change, 
most of the time options be used in engine only. May be it makes sense to move it to engine rather than keeping it in the extension.